File: multiping.py
import re
import subprocess

sites = ['google.com','perdu.com','laredoute.fr','yahoo.com','xkcd.com','arstechnica.com','www.wired.com','sfr.fr','actu.fr','wanadoo.fr']
results = []

# Extracting data from ping and putting it in an array of tuples
for el in sites:
	cmd = "ping -n 1 " + el
	output = subprocess.check_output(cmd)
	output = ''.join(map(chr, output))
	latencies = re.search(r"([0-9]+)ms\r\n$", output)
	results.append((el, latencies.group(1)))

# Sorting the array by the latency of each site
results.sort(key = lambda x: float(x[1]), reverse = False)

# Displaying the result
for i in results:
	print(i[0], "=", i[1], "ms")

File: multiping.rb
require 'open3'
require 'ostruct'

sites = ['google.com','perdu.com','laredoute.fr','yahoo.com','xkcd.com','arstechnica.com','www.wired.com','sfr.fr','actu.fr','wanadoo.fr']
results = []

for el in sites do
	Open3.popen3("ping -n 1 " + el) do |stdin, stdout, stderr|
		# This is the only way to not have the error "in `[]': invalid byte sequence in UTF-8 (ArgumentError)"
		@output = stdout.read.force_encoding("ISO-8859-1").encode('utf-8')
	end
	# Extract only the second group which we are interested in
	latency = @output[/(Moyenne = )([0-9]+)(ms)/, 2]
	tuple = OpenStruct.new
	tuple.site = el
	tuple.latency = latency.to_i
	results.push(tuple)
end
# Sorting by latency
results = results.sort_by { |i| i.latency}

# Displaying the result
for el in results do
	puts el.site + ' = ' + el.latency.to_s + ' ms'
end